drivers/backup/portworx/portworx.go

```diff
@@ -0,0 +1,349 @@
+	p.cloudCredentialManager = api.NewCloudCredentialClient(conn)
+	p.backupManger = api.NewBackupClient(conn)
+	p.restoreManager = api.NewRestoreClient(conn)
+	p.backupScheduleManager = api.NewBackupScheduleClient(conn)
+	p.schedulePolicyManager = api.NewSchedulePolicyClient(conn)
+	p.organizationManager = api.NewOrganizationClient(conn)
+
+	logrus.Infof("Using %v as endpoint for portworx backup driver", pxEndpoint)
+
+	return err
+}
+
+func (p *portworx) GetServiceEndpoint(serviceName string, nameSpace string) (string, error) {
+	svc, err := core.Instance().GetService(serviceName, nameSpace)
+	if err == nil {
+		return svc.Spec.ClusterIP, nil
+	}
+	return "", err
+}
+
+func (p *portworx) setDriver(serviceName string, nameSpace string) error {
+	var err error
+	var endpoint string
+
+	endpoint, err = p.GetServiceEndpoint(serviceName, nameSpace)
+	if err == nil && endpoint != "" {
+		if err = p.testAndSetEndpoint(endpoint); err == nil {
+			p.refreshEndpoint = false
+			return nil
+		}
+	} else if err != nil && len(node.GetWorkerNodes()) == 0 {
+		return err
+	}
+
+	p.refreshEndpoint = true
+	for _, n := range node.GetWorkerNodes() {
+		for _, addr := range n.Addresses {
+			if err = p.testAndSetEndpoint(addr); err == nil {
+				return nil
+			}
+		}
+	}
+
+	return fmt.Errorf("failed to get endpoint for portworx backup driver: %v", err)
+}
+
+func (p *portworx) getOrganizationManager() api.OrganizationClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.organizationManager
+}
+
+func (p *portworx) getClusterManager() api.ClusterClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.clusterManager
+}
+
+func (p *portworx) getBackupLocationManager() api.BackupLocationClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.backupLocationManager
+}
+
+func (p *portworx) getCloudCredentialManager() api.CloudCredentialClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.cloudCredentialManager
+}
+
+func (p *portworx) getBackupManager() api.BackupClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.backupManger
+}
+
+func (p *portworx) getRestoreManager() api.RestoreClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.restoreManager
+}
+
+func (p *portworx) getBackupScheduleManager() api.BackupScheduleClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.backupScheduleManager
+}
+
+func (p *portworx) getSchedulePolicyManager() api.SchedulePolicyClient {
+	if p.refreshEndpoint {
+		p.setDriver(pxbServiceName, pxbNameSpace)
+	}
+	return p.schedulePolicyManager
+}
+
+func (p *portworx) CreateOrganization(req *api.OrganizationCreateRequest) (*api.OrganizationCreateResponse, error) {
+	org := p.getOrganizationManager()
+	if req != nil {
```

Comment:
I don't believe you need this check, the API will return an error if the request is nil.

Reply:
fixed